src/cosmic_ray/tools/survival_rate.py

```diff
@@ -8,16 +10,65 @@
 def format_survival_rate():
     """cr-rate
 
-    Usage: cr-rate <session-file>
+Usage: cr-rate [--estimate] [--confidence <confidence_rate>] [--fail-over <max_value>] <session-file>
 
-    Calculate the survival rate of a session.
-    """
+Calculate the survival rate of a session.
+
+options:
+    --estimate                     Print the lower bound, estimate and upper bound of
+                                   survival rate
+    --confidence <confidence_rate> Specify the confidence levels for estimates, 95 (%) by
+                                   default. 80, 90, 95, 98, 99, 99.5, 99.8 and 99.9 are
+                                   supported.
+    --fail-over <max_value>        Exit with a non-zero code if the survival rate is
+                                   larger than <max_value> or the calculated confidence
+                                   interval is above the <max_value> (if --estimate is used).
+                                   Specified as percentage.
+"""
     arguments = docopt.docopt(
         format_survival_rate.__doc__, version='cr-rate 1.0')
+    show_estimate = arguments['--estimate']
+    confidence = arguments['--confidence']
+    fail_over = arguments['--fail-over']
+
+    # use integers as keys as equality is not well defined for floats
```

Comment:
I'm a bit wary of this; floating point equality is perfectly well defined, though I suspect you're (reasonably) concerned about rounding causing confusion. If we can't use floats and can't calculate z-values for arbitrary confidence intervals, would an enum be more appropriate?

Reply:
> I'm a bit wary of this; floating point equality is perfectly well defined,

hmm, since we're not calculating it, but just comparing hard-coded values, yes, equality would work

> If we can't use floats and can't calculate z-values for arbitrary confidence intervals

sure, it is possible to calculate it, but as far as I can tell, it would require integration of a not exactly trivial function so I don't think it's really worth it

> would an enum be more appropriate?

if the confidence intervals calculation was to be supported for other values, and the code generalised, probably; I do think that `dict()` aligns better with a "lookup table" than an enum though, and in the future we may need to add 99.99 or some-such